components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java

```diff
@@ -398,19 +482,96 @@ public boolean removeTopic(String topicName) throws EventBrokerException {
             String resourcePath = JavaUtil.getResourcePath(topicName, this.topicStoragePath);
             return userRegistry.resourceExists(resourcePath);
         } catch (RegistryException e) {
-            throw new EventBrokerException("Can not access the config registry");
+            throw new EventBrokerException("Cannot access the config registry");
+        }
+    }
+
+    /**
+     * Create a new role which has the same name as the destinationName and assign the logged in
+     * user to the newly created role. Then, authorize the newly created role to subscribe and
+     * publish to the destination.
+     *
+     * @param username        name of the logged in user
+     * @param destinationName destination name. Either topic or queue name
+     * @param destinationId   ID given to the destination
+     * @param userRealm       the  user store
+     * @throws UserStoreException
+     */
+    private static void authorizePermissionsToLoggedInUser(String username, String destinationName,
+                                                           String destinationId,
+                                                           UserRealm userRealm) throws
+                                                                                UserStoreException {
+
+        //For registry we use a modified queue name
+        String newDestinationName = destinationName.replace("@", AT_REPLACE_CHAR);
+
+        // creating the internal role name
+        String roleName = UserCoreUtil.addInternalDomainName(TOPIC_ROLE_PREFIX +
+                                                             newDestinationName.replace("/", "-"));
+
+        // the interface to store user data
+        UserStoreManager userStoreManager = CarbonContext.getThreadLocalCarbonContext().getUserRealm().getUserStoreManager();
+
+        if (!userStoreManager.isExistingRole(roleName)) {
+            String[] user = {MultitenantUtils.getTenantAwareUsername(username)};
+
+            // adds the internal role to user store
+            userStoreManager.addRole(roleName, user, null);
+            // gives subscribe permissions to the internal role in the user store
+            userRealm.getAuthorizationManager().authorizeRole(
+                    roleName, destinationId, EventBrokerConstants.EB_PERMISSION_SUBSCRIBE);
+            // gives publish permissions to the internal role in the user store
+            userRealm.getAuthorizationManager().authorizeRole(
+                    roleName, destinationId, EventBrokerConstants.EB_PERMISSION_PUBLISH);
+            // gives change permissions to the internal role in the user store
+            userRealm.getAuthorizationManager().authorizeRole(
+                    roleName, destinationId, EventBrokerConstants.EB_PERMISSION_CHANGE_PERMISSION);
+
+        } else {
+            log.warn("Unable to provide permissions to the user, " +
```

Comment:
Yes, It should be thrown.
Fixed in https://github.com/wso2/carbon-commons/pull/66.